bids/grabbids/bids_layout.py

```diff
@@ -124,12 +121,60 @@ def get_bval(self, path, **kwargs):
         else:
             return tmp
 
-    def get_events(self, path, **kwargs):
-        tmp = self._get_nearest_helper(path, '.tsv', type='events', **kwargs)
-        if isinstance(tmp, list):
-            return tmp[0]
+    def get_events(self, path, output='file', derivatives='both', **kwargs):
+        """ For a given file in a BIDS project, finds corresponding event files
+        and optionally returns merged dataframe containing all variables.
+
+        Args:
+            path (string): Path to a file
+            output (string): Output to return.
+                'file' returns list of files, 'df' returns merged dataframe.
+            derivatives (string): How to handle derivative events. One of
+                'ignore', 'both', or 'only'
+            how (string): Merge method.
+        Returns:
+            List of file or merged Pandas dataframe.
+        """
+
+        path = abspath(path)
+
+        # Get events in base Layout directory (ordered)
+        root_events = self._get_nearest_helper(
+            path, '.tsv', type='events', **kwargs) or []
+
+        entities = self.files[path].entities
+        if 'type' in entities:
+            entities.pop('type')
+        entities.update(kwargs)
+
+        # Get all events
+        events =  self.get(extensions='tsv', type='events',
+                           return_type='file', **entities) or []
+
+        deriv_events = list(set(events) - set(root_events))
+
+        if derivatives == 'only':
+            events = deriv_events
+        elif derivatives == 'ignore':
+            events = root_events
+        else: # Combine with order
+            events = deriv_events + root_events
+
+        if output == 'df':
+            # Merge dataframes sequentially, giving preference to values from
+            # event files higher in the hierarchy
+            merged = None
+            for e in events:
+                e = pd.read_csv(e, delimiter='\t')
```

Comment:
Probably better to move the reading into utils as well--that way we don't need pandas here at all. We can have a utility called `_merge_event_files`.